### ISSUE TITLE
Lammps/quip 

### ISSUE THREAD

**Issue by ZengZezhu** [libAtoms/QUIP#217]
Hi, professor, I encountered  a very strange problem when running the lammps MD by GAPs/quip.

I used the GAPs to fit a potential and used the quip (installed in my laptop) to calculate the forces of a random structure , then made a comparison with the VASP forces results, which is agreed very well. 

However, if I used the same structure to calculate the forces by Lammps/quip in supercomputer, then there is a large difference between lammps forces and VASP forces.

So my question is that the forces from lammps/quip is differ compared with the forces from myself quip, however, there is no error when I installed lammps with quip in the supercomputer. So I am so confused about this problem.

My lammps version is LAMMPS (3 Mar 2020), and I just obtained the quip directly by the git clone --recursive https://github.com/libAtoms/QUIP.git

Best,
zzz

**Comment by gabor1** [comment]
Lammps has choice of units, did you set that correctly? 

-- Gábor

Gábor Csányi
Professor of Molecular Modelling
Engineering Laboratory, University of Cambridge
Pembroke College Cambridge

Pembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/

**Comment by ZengZezhu** [comment]
I used :

units           metal
boundary        p p p

atom_style      atomic

**Comment by ZengZezhu** [comment]
I think it is the version problem of lammps.

**Comment by ZengZezhu** [comment]
Hi professor,

Could you provide some suggestions about the version of lammps? My GAP version is gap_version="1591702871">.
I just reinstalled the lammps/quip by changing the version of lammps to LAMMPS (31 Mar 2017), and there is a error like below:
Segmentation fault (core dumped)

**Comment by albapa** [comment]
What was your QUIP_ARCH?

**Comment by albapa** [comment]
I've checked a fresh install of LAMMPS/QUIP and they seem to be consistent. What was your system? Are your atom types  mapped correctly?

**Comment by ZengZezhu** [comment]
My system is linux and I just loaded these software:

1) openssl/1.1.1b   2) python/3.8.3   3) gcc/6.3.0   4) mpich/gcc/3.1.4

I downloaded the lammps-3Mar20 and checked the README file in lammps-3Mar20/lib/quip, then I downloaded the QUIP by:
''git clone https://github.com/libAtoms/QUIP.git QUIP''  in lammps-3Mar20/lib/quip directory

As I also want to use the GAPs, so I downloaded the GAPs (version="1591702871") into the QUIP/src.

Then I did these operations to build the quip:

cd QUIP
$ export QUIP_ROOT=${PWD}
$ export QUIP_ARCH=linux_x86_64_gfortran
$ make config (GAP is ok)
$ make libquip

There is no error to occur.

Then I try to build the lammps in lammps-3Mar20/

1. make yes-user-quip
2. go to the lammps/src and make -j 4 mpi

there is also no error to occur.

I generated the lmp.data from the ASE, so the input data seems has no problem, and the lammps will have a mistake if the atom types mapping is wrong.

Could you tell me the load environment and version of lammps, quip and GAPs? I will try again.

**Comment by ZengZezhu** [comment]
Hi, I solved this problem by using lammps-24Oct2018. :)
But I dont know the intrinsic problem.:)

**Comment by gabor1** [comment]
ok, now we know what to diff... our interface needs updating. 

-- Gábor

**Comment by ZengZezhu** [comment]
Thanks!

**Comment by albapa** [comment]
@ZengZezhu do you have a minimal example input where LAMMPS is inconsistent with standalone quip?

**Comment by ZengZezhu** [comment]
Sorry I have no minimal example, I just deleted the lammps-3Mar20. I just
used a random structure to check the forces from standalone quip and
lammps/quip.
But I think if it is an intrinsic problem then maybe the testing result
does not depend on the system.

**Comment by ZengZezhu** [comment]
And maybe it is my problem when installing the software or loading some
modules.

**Comment by ZengZezhu** [comment]
Hi, professor,

I just found that the cur_decomposition iteration error is dependent with
values of n_sparse, l_max and n_max. I dont understand the meaning of this
parameter.

But I have an intuitive opinion that if I increase the degree of complexity
of a potential model, the error will be decreased. Could you give me an
intuitive judgement about this? In other words, should I go on decreasing
this error if it reaches 10^{-5}.

Best,
zzz

**Comment by gabor1** [comment]
I wouldn’t worry about that at all. it’s just the residual of the linear algebra. if it’s large, like O(1), then report it!

(maybe we should put that message there)

-- Gábor

Gábor Csányi
Professor of Molecular Modelling
Engineering Laboratory, University of Cambridge
Pembroke College Cambridge

Pembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/

**Comment by ZengZezhu** [comment]
Ok, thanks a lot!

**Comment by ZengZezhu** [comment]
Hi, Professor.

I used the fitted potential from GAPs to run the MD by lammps and the result is strange.

I fitted the potential by some 300K-500K configurations from AIMD and the fitting line is:

# gap_fit energy_parameter_name=energy force_parameter_name=forces virial_parameter_name=stress do_copy_at_file=F sparse_separate_file=T gp_file=GAP.xml at_file=train.xyz default_sigma={0.001 0.03 0.03 0} gap={distance_2b cutoff=7.0 covariance_type=ard_se delta=0.5 theta_uniform=1.0 sparse_method=uniform add_species=T n_sparse=20 : soap l_max=6 n_max=6 cutoff=6.0 cutoff_transition_width=1.0 delta=1.0 atom_sigma=0.5 zeta=4 sparse_method=cur_points  add_species=T n_sparse=500 covariance_type=dot_product}

The testing RMSE of energy and force are 0.001eV and 0.04eV/Ang, respectively. 

#Then I used this potential to run the MD by lammps. This is my script of lammps:

units		metal
boundary	p p p

atom_style	atomic

read_data	lmp.data

pair_style	quip
pair_coeff	* * GAP.xml "Potential xml_label=GAP_2020_7_1_480_14_25_49_518" 56 47 52

neighbor	0.0 bin
neigh_modify	delay 10

variable t equal 300
velocity all create $t 12345 mom yes rot yes dist gaussian

fix NVT all nvt temp $t $t 0.1
fix   m_zero all momentum 100 linear 1 1 1 

run 20000

unfix NVT
fix   NVE all nve

thermo   100 
thermo_modify   flush yes
dump         dynaphopy all custom 5 dump.lammpstrj vx vy vz
dump_modify  dynaphopy sort id
run      500000

Question:

I just found that the total energy in NVE ensemble is decreased monotonically with the running time.
[tot_e.pdf](https://github.com/libAtoms/QUIP/files/4874701/tot_e.pdf)

It is so confused to me because the total energy should not be changed more in NVE ensemble. 

I don't know whether the configurations have some effects to the MD. I used the cubic cell to fit the potential and however, in MD simulation is the oblique supercell.

If you need more information about this problem, please contact me without any hesitation.

Best,
zzz

**Comment by gabor1** [comment]
this is very strange. even if the potential was completely wrong, as you say nve should never change the total energy, especially not linearly like this. can you please try to run a short md using the "md" program shipped with QUIP ? "md —help" gives simple command line options. If you don’t give a temperature, it defaults to NVE. 

-- Gábor

Gábor Csányi
Professor of Molecular Modelling
Engineering Laboratory, University of Cambridge
Pembroke College Cambridge

Pembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/

**Comment by ZengZezhu** [comment]
Sorry I dont know how to use the quip to run the MD because I had
not installed the quippy.  Could you give a simple command line to me to
run the MD?

I try to run the md -help but there seems to be no useful information....

**Comment by ZengZezhu** [comment]
you mean I also run the md by quip by command line like gaps?

**Comment by ZengZezhu** [comment]
I see, I will try.

**Comment by gabor1** [comment]
I didn’t mean quippy (the python interface), but the command line program "md" that you get if you type "make" in the QUIP directory. 

-- Gábor

Gábor Csányi
Professor of Molecular Modelling
Engineering Laboratory, University of Cambridge
Pembroke College Cambridge

Pembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/

**Comment by ZengZezhu** [comment]
I see, thanks professor. Actually I also can use the quippy and use the ASE to run the md, it is so convenient.
This is my result when I used the quippy and ASE to run the NVE md. First, I need to clarify that I used a smaller oblique supercell to run the md compared with lammps as the larger supercell needs more time to get the results in my laptop.
I found that the total energy is constant when used the quippy with ASE like below:
[e_tot_each_atom.pdf](https://github.com/libAtoms/QUIP/files/4876523/e_tot_each_atom.pdf)

So maybe it is the problem from lammps?
